src/jit/codegencommon.cpp

```diff
@@ -2674,15 +2674,22 @@ void            CodeGen::genJumpToThrowHlpBlk(emitJumpKind          jumpKind,
         /* The code to throw the exception will be generated inline, and
            we will jump around it in the normal non-exception case */
 
-        BasicBlock * tgtBlk = genCreateTempLabel();
-        jumpKind = emitter::emitReverseJumpKind(jumpKind);
-        inst_JMP(jumpKind, tgtBlk);
+        BasicBlock * tgtBlk = nullptr;
+        emitJumpKind reverseJumpKind = emitter::emitReverseJumpKind(jumpKind);
+        if (reverseJumpKind != jumpKind)
```

Comment:
An observation:  This looks to be a bug even in codegenlegacy.cpp, since it is calling this method with EJ_jmp (jump always) in one instance in which array length and indiex is a constant and there is a bounds check failure.

Probably the reason why this wasn't exposed so far is this method wasn't getting exercised with EJ_jmp on RyuJIT xarch so far.  Arm64 back-end is the first one doing it for div-by-zero case.